FreeRTOS/FreeRTOS
# Add Atmel Studio projects for ATMega4809 and AVR128DA48

Added Atmel Studio projects for ATMega4809 and AVR128DA48

Description
-----------
Beside the projects there are a couple of changes in Demo/Common files. These changes are:
o	(1) In semtest.h I updated the return type of xAreSemaphoreTasksStillRunning function from BaseType_t to portBASE_TYPE because this is the return type it has where it is implemented (semtest.c).
o	(2) In recmutex.c I added an explicit cast to UBaseType_t On line 260. I did this because the right-hand expression in the “==” check was, at some point, expected to roll over and therefore be equal to the left-hand side (that was, at that time, 0). Instead, what I think was happening was that the compiler performed integer promotion and turned the 8 bit variable into a 32 bit one, this way roll over did not happened and it was actually comparing 0 to 256 instead of 0 to 255 + 1 (=0 because of roll over).

Test Steps
-----------
From what I can tell the above changes should be backwards compatible because:  in change (1) the two types actually resolve to the same primitive type and in change (2) the cast actually serves as an additional clarification to the "compiler" about what the code is expected to do.
But you might have a bigger picture than me, and have some more tests to perform in order to confirm the changes are ok, so this is way I described them in detail.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
